menas/src/main/scala/za/co/absa/enceladus/menas/controllers/PropertyDefinitionController.scala

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2018 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.menas.controllers
+
+import java.net.URI
+import java.util.concurrent.CompletableFuture
+
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.http.{HttpStatus, ResponseEntity}
+import org.springframework.security.core.annotation.AuthenticationPrincipal
+import org.springframework.security.core.userdetails.UserDetails
+import org.springframework.web.bind.annotation._
+import za.co.absa.enceladus.menas.services.PropertyDefinitionService
+import za.co.absa.enceladus.model.properties.PropertyDefinition
+
+import scala.concurrent.ExecutionContext.Implicits.global
+
+@RestController
+@RequestMapping(path = Array("/api/properties/datasets"), produces = Array("application/json"))
+class PropertyDefinitionController @Autowired()(propertyDefService: PropertyDefinitionService)
+  extends VersionedModelController(propertyDefService) {
+
+  import za.co.absa.enceladus.menas.utils.implicits._
+
+  @GetMapping(Array(""))
+  def getAllDatasetProperties(): CompletableFuture[Seq[PropertyDefinition]] = {
+    logger.info("retrieving all dataset properties in full")
+    propertyDefService.getLatestVersions
+  }
+
+  @PostMapping(Array(""))
+  @ResponseStatus(HttpStatus.CREATED)
+  def createDatasetProperty(@AuthenticationPrincipal principal: UserDetails,
+                            @RequestBody item: PropertyDefinition): CompletableFuture[ResponseEntity[PropertyDefinition]] = {
+    // basically an alias for /create with Location header response
+    logger.info(s"creating new property definition '${item.name}'")
+
```

Comment:
This passes while it should not. SuggestedValue does not exist in the allowedValues

```json
{
	"name": "firstProp",
  "propertyType": {
		"_t" : "EnumPropertyType ",
		"allowedValues": ["optionA", "optionB", "optionC"],
		"suggestedValue" : "optionD",
		"putIntoInfoFile" : false,
		"essentiality" : { "_t" : "Optional" }
	}
}
```

Reply:
Even though I believe that the correct payload should be
```
{
    "name": "myPdKey2",
    "description": "myDesc",
    "propertyType": {
        "_t": "EnumPropertyType ",
        "allowedValues": ["optionA", "optionB", "optionC"],
        "suggestedValue": "optionD"
    },
    "essentiality": {
        "_t": "Optional"
    },
    "putIntoInfoFile" : false
}
```
(`allowedValues` and `suggestedValue` are part of the `propertyType`, while the rest of field is related to the root property definition instead), you are correct in the fact, that the suggestedValue should be checked while it is probably not. I will revisit this and test-cover it, too.

Reply:
IMHO Fixed now, I have added some tests to cover the case.